Doc: Add json_lines known issue to release notes (#16831)
elastic/logstash: docs/static/releasenotes.asciidoc

=== Logstash 8.17.0 Release Notes

[[known-issues-8-17-0]]
==== Known issues

[[known-issue-8-17-0-jvm]]
===== JVM version changes needed when upgrading {ls} from 8.12.0 (or earlier)

If the `jvm.options` file was modified and not overwritten with the newest version, you may see a "deserialize invocation error" message, causing the pipeline to crash. 
Users are affected if the Persistent Queue (PQ) is enabled, and the pipeline is processing messages larger than 20MB.

> **Solution:** Apply the default change contained in the newer 'jvm.options' file, as seen in this https://github.com/elastic/logstash/blob/main/config/jvm.options#L66-L82[example].

Reviewer: I understand this was just an in-place refactor, but we should link to the released tag's docs (which are static) instead of the docs on `main` (which is ever-evolving)

```suggestion
**Solution:** Apply the default change contained in the newer 'jvm.options' file, as seen in this https://github.com/elastic/logstash/blob/v8.17.0/config/jvm.options#L74-L90[example].
```

```
unable to process event. {:message=>"input buffer full", :class=>"Java::JavaLang::IllegalStateException", :backtrace=>["org.logstash.common.BufferedTokenizerExt.extract(BufferedTokenizerExt.java:83)", "usr.share.logstash.vendor.bundle.jruby.$3_dot_1_dot_0.gems.logstash_minus_codec_minus_json_lines_minus_3_dot_2_dot_2.lib.logstash.codecs.json_lines.RUBY$method$decode$0(/usr/share/logstash/vendor/bundle/jruby/3.1.0/gems/logstash-codec-json_lines-3.2.2/lib/logstash/codecs/json_lines.rb:69)", "org.jruby.internal.runtime.methods.CompiledIRMethod.call(CompiledIRMethod.java:165)", "org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:185)",
```
The issue was fixed in https://github.com/elastic/logstash/pull/16760.

This problem is most likely to be seen when you are using the <<plugins-integrations-logstash,{ls} integration>> plugin to ship data between two {ls} instances, but may appear in other situations, too. 

**Workaround for {ls}-to-{ls} communication** 

The {ls}-to-{ls} issue can be mitigated by:

* Downgrading the _receiving_ {ls} to `8.16.2`, or any {ls} in the `8.15` series,  **_AND/OR_**

> * Upgrading the {ls} integration filter of the _sending_ {ls} to version `1.0.4`.

Author: ```suggestion
* Upgrading the <<plugins-integrations-logstash,{ls} integration>> plugin of the _sending_ {ls} to version `1.0.4`.
```

This should be the Logstash integration plugin, right?  The agent_integration filter doesn't make sense to me.  

Note to self @karenzone:  Make sure any change here gets propogated to 8.16.1 and 8.16.0 sections.

Reviewer 2: Correct - it's the logstash input and output that is affected.

=== Logstash 8.17.0 Release Notes

[[known-issues-8-17-0]]
==== Known issues

[[known-issue-8-17-0-jvm]]
===== JVM version changes needed when upgrading Logstash from 8.12.0 (or earlier)

If the `jvm.options` file was modified and not overwritten with the newest version, you may see a "deserialize invocation error" message, causing the pipeline to crash. 
Users are affected if the Persistent Queue (PQ) is enabled, and the pipeline is processing messages larger than 20MB.

**Solution:** Apply the default change contained in the newer 'jvm.options' file, as seen in this https://github.com/elastic/logstash/blob/main/config/jvm.options#L66-L82[example].

[[known-issue-8-17-0-json_lines]]
===== "Input buffer full" error with {ls} 8.16.0, 8.16.1, or 8.17.0

> If you are using `json_lines` codec 3.2.0 (or earlier) with {ls} 8.16.0, 8.16.1, or 8.17.0, you may see an error similar to this one, crashing the pipelines:

Reviewer 2: ```suggestion
If you are using `json_lines` codec 3.2.0 (or later) with {ls} 8.16.0, 8.16.1, or 8.17.0, you may see an error similar to this one, crashing the pipelines:
```

Reviewer 2: I wonder if we need to mention the version of the `json_lines` codec, as it is implied by the version of Logstash?

Reviewer 2: (I'm trying to think of ways we can simplify the message)

=== Logstash 8.17.0 Release Notes

[[known-issues-8-17-0]]
==== Known issues

[[known-issue-8-17-0-jvm]]
===== JVM version changes needed when upgrading Logstash from 8.12.0 (or earlier)

If the `jvm.options` file was modified and not overwritten with the newest version, you may see a "deserialize invocation error" message, causing the pipeline to crash. 
Users are affected if the Persistent Queue (PQ) is enabled, and the pipeline is processing messages larger than 20MB.

**Solution:** Apply the default change contained in the newer 'jvm.options' file, as seen in this https://github.com/elastic/logstash/blob/main/config/jvm.options#L66-L82[example].

[[known-issue-8-17-0-json_lines]]

> ===== Error with `json_lines` codec 3.2.0 (or earlier) and {ls} 8.17.1 (or earlier)

Reviewer 3: ```suggestion
===== Error with `json_lines` codec 3.2.0 (or earlier) with {ls} 8.17.0
```
`and {ls} 8.17.1 (or earlier)` - part is a bit confusing to me. I support removing LS version from this title which brings reader to "... present in {ls} versions `8.16.0`, `8.16.1`, and `8.17.0`." and to mitigation section.

Author: This was based on description in issue: "When using the json_lines codec > 3.2.0 with versions of **_Logstash < 8.17.1_**, the following error may be emitted, crashing the pipelines"

If that's not accurate, I'll update it here, and it will also need to be updated in the issue.

Reviewer 2: What do you think about using the _symptom_ as the heading, rather than `json_lines` codec?

```
===== "Input Buffer Full" error With Logstash `8.17.0`, `8.16.1` and `8.16.0`
```

Then describe the instances of how this would happen and how to mitigate, with Logstash->Logstash communication first, as it is the most likely, then the  `json_lines` case?

Leading with `json_lines` isn't necessarily obvious as the codec is used "silently" by the Logstash integration plugin.

> ===== Error with `json_lines` codec 3.2.0 (or earlier) and {ls} 8.17.1 (or earlier)

Author: ```suggestion
===== Error with `json_lines` codec 3.2.0 (or later) and {ls} 8.17.1 (or earlier)
```
Based on description in the issue, I think this should be 3.2.0 or later for json_lines?

Reviewer 3: Yup, `json_lines codec >= 3.2.0` versions which have an ability to set the buffer limit.

[[logstash-8-16-1]]
=== Logstash 8.16.1 Release Notes

[[known-issues-8-16-1]]
==== Known issue

[[known-issue-8-16-1-json_lines]]

> ===== "Input buffer full" error with {ls} 8.16.0, 8.16.1, or 8.17.0

Author: It seems kinda weird to have 8.17.0 listed in the 8.16.1 and 8.16.0 release notes.  
However, this information is complete and correct, and I'm inclined to leave it. LMKWYT.

Note to self @karenzone: Make sure any change here gets propagated to 8.16.1 and 8.16.0 sections.

Reviewer 2: It does, and it's not something we typically do. 

However, the fact that this is not affected in `8.16.2` may make this appropriate to be explicit about which versions are affected to ensure that users are not confused, and don't think we have just missed `8.16.2`.

It's something that could affect us in the future, as we support multiple branches, and the bug story isn't as simple as it used to be, so it's probably up for debate how we want to approach this from a messaging standpoint.
